### ISSUE TITLE
redmine_crm update fails

### ISSUE THREAD

**Issue by Lokizzle** [rubygems/rubygems#6262]
### Describe the problem as clearly as you can

I´m running redmine 4.2.3 on RHEL 8 OS.

Running redmine
Environment:
  Redmine version                4.2.3.stable
  Ruby version                   2.5.9-p229 (2021-04-05) [x86_64-linux]
  Rails version                  5.2.6

I wanted to install 2x new plugins redmine_checklists and redmine_agile. 
When im trying to do so by hitting the command:
 `bundle exec rake redmine:plugins NAME=redmine_checklists RAILS_ENV=production`
I get this error:
Your RubyGems version (2.7.6.3)) has a bug that prevents `required_ruby_version` from working for Bundler. Any scripts that use `gem install bundler` will break as soon as Bundler drops support for your Ruby version. Please upgrade RubyGems to avoid future breakage and silence this warning by running `gem update --system 3.2.3`
Redmine requires redmine_crm gem version 0.0.56 or higher (you're using 0.0.55).
Please update with 'bundle update redmine_crm'.

### Did you try upgrading rubygems & bundler?

if i try bundle update redmine_crm i get this error:

redmine-4.2.3_new]$ bundle update redmine_crm
Your RubyGems version (2.7.6.3)) has a bug that prevents `required_ruby_version` from working for Bundler. Any scripts that use `gem install bundler` will break as soon as Bundler drops support for your Ruby version. Please upgrade RubyGems to avoid future breakage and silence this warning by running `gem update --system 3.2.3`
Your Gemfile lists the gem redmine_crm (>= 0) more than once.
You should probably keep only one of them.
Remove any duplicate entries and specify the gem only once.
While it's not a problem now, it could cause errors if you change the version of one of them later.
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/versions.rb:5: warning: already initialized constant Psych::VERSION
/usr/share/ruby/psych/versions.rb:4: warning: previous definition of VERSION was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:7: warning: already initialized constant Psych::ClassLoader::BIG_DECIMAL
/usr/share/ruby/psych/class_loader.rb:7: warning: previous definition of BIG_DECIMAL was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:8: warning: already initialized constant Psych::ClassLoader::COMPLEX
/usr/share/ruby/psych/class_loader.rb:8: warning: previous definition of COMPLEX was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:9: warning: already initialized constant Psych::ClassLoader::DATE
/usr/share/ruby/psych/class_loader.rb:9: warning: previous definition of DATE was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:10: warning: already initialized constant Psych::ClassLoader::DATE_TIME
/usr/share/ruby/psych/class_loader.rb:10: warning: previous definition of DATE_TIME was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:11: warning: already initialized constant Psych::ClassLoader::EXCEPTION
/usr/share/ruby/psych/class_loader.rb:11: warning: previous definition of EXCEPTION was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:12: warning: already initialized constant Psych::ClassLoader::OBJECT
/usr/share/ruby/psych/class_loader.rb:12: warning: previous definition of OBJECT was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:13: warning: already initialized constant Psych::ClassLoader::PSYCH_OMAP
/usr/share/ruby/psych/class_loader.rb:13: warning: previous definition of PSYCH_OMAP was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:14: warning: already initialized constant Psych::ClassLoader::PSYCH_SET
/usr/share/ruby/psych/class_loader.rb:14: warning: previous definition of PSYCH_SET was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:15: warning: already initialized constant Psych::ClassLoader::RANGE
/usr/share/ruby/psych/class_loader.rb:15: warning: previous definition of RANGE was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:16: warning: already initialized constant Psych::ClassLoader::RATIONAL
/usr/share/ruby/psych/class_loader.rb:16: warning: previous definition of RATIONAL was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:17: warning: already initialized constant Psych::ClassLoader::REGEXP
/usr/share/ruby/psych/class_loader.rb:17: warning: previous definition of REGEXP was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:18: warning: already initialized constant Psych::ClassLoader::STRUCT
/usr/share/ruby/psych/class_loader.rb:18: warning: previous definition of STRUCT was here
/usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:19: warning: already initialized constant Psych::ClassLoader::SYMBOL
/usr/share/ruby/psych/class_loader.rb:19: warning: previous definition of SYMBOL was here
--- ERROR REPORT TEMPLATE -------------------------------------------------------

```
SyntaxError: (eval):2: syntax error, unexpected =>, expecting '}'
  load {"Complex"=>Complex, "Regexp"=>Regexp, "E...
         ^~
(eval):2: dynamic constant assignment
  load {"Complex"=>Complex, "Regexp"=>Regexp, "Exception...
                  ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
...{"Complex"=>Complex, "Regexp"=>Regexp, "Exception"=>Exceptio...
...                             ^~
(eval):2: dynamic constant assignment
...ex"=>Complex, "Regexp"=>Regexp, "Exception"=>Exception, "Obj...
...                              ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
..."Regexp"=>Regexp, "Exception"=>Exception, "Object"=>Object, ...
...                             ^~
(eval):2: dynamic constant assignment
...Regexp, "Exception"=>Exception, "Object"=>Object, "Psych::Om...
...                              ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
...ception"=>Exception, "Object"=>Object, "Psych::Omap"=>Psych:...
...                             ^~
(eval):2: dynamic constant assignment
..."=>Exception, "Object"=>Object, "Psych::Omap"=>Psych::Omap, ...
...                              ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
...bject"=>Object, "Psych::Omap"=>Psych::Omap, "Range"=>Range, ...
...                             ^~
(eval):2: dynamic constant assignment
...ct, "Psych::Omap"=>Psych::Omap, "Range"=>Range, "Rational"=>...
...                              ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
...::Omap"=>Psych::Omap, "Range"=>Range, "Rational"=>Rational, ...
...                             ^~
(eval):2: dynamic constant assignment
..."=>Psych::Omap, "Range"=>Range, "Rational"=>Rational, "Psych...
...                              ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
...p, "Range"=>Range, "Rational"=>Rational, "Psych::Set"=>Psych...
...                             ^~
(eval):2: dynamic constant assignment
..."=>Range, "Rational"=>Rational, "Psych::Set"=>Psych::Set, "S...
...                              ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
...onal"=>Rational, "Psych::Set"=>Psych::Set, "Struct"=>Struct,...
...                             ^~
(eval):2: dynamic constant assignment
...onal, "Psych::Set"=>Psych::Set, "Struct"=>Struct, "Symbol"=>...
...                              ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
...h::Set"=>Psych::Set, "Struct"=>Struct, "Symbol"=>Symbol}
...                             ^~
(eval):2: dynamic constant assignment
...=>Psych::Set, "Struct"=>Struct, "Symbol"=>Symbol}
...                              ^
(eval):2: syntax error, unexpected =>, expecting &. or :: or '[' or '.'
...t, "Struct"=>Struct, "Symbol"=>Symbol}
...                             ^~
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:38:in `class_eval'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:38:in `block in <class:ClassLoader>'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:36:in `each'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:36:in `<class:ClassLoader>'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:6:in `<module:Psych>'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/class_loader.rb:5:in `<top (required)>'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/nodes/node.rb:3:in `require_relative'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/nodes/node.rb:3:in `<top (required)>'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/nodes.rb:2:in `require_relative'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych/nodes.rb:2:in `<top (required)>'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych.rb:15:in `require_relative'
  /usr/local/share/gems/gems/psych-4.0.3/lib/psych.rb:15:in `<top (required)>'
  /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:59:in `require'
  /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:59:in `require'
  /usr/share/rubygems/rubygems.rb:694:in `load_yaml'
  /usr/share/rubygems/rubygems/config_file.rb:349:in `load_file'
  /usr/share/rubygems/rubygems/config_file.rb:202:in `initialize'
  /usr/share/rubygems/rubygems.rb:350:in `new'
  /usr/share/rubygems/rubygems.rb:350:in `configuration'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/fetcher.rb:243:in `connection'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/fetcher.rb:90:in `initialize'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/source/rubygems.rb:288:in `new'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/source/rubygems.rb:288:in `block in fetchers'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/source/rubygems.rb:286:in `map'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/source/rubygems.rb:286:in `fetchers'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/source/rubygems.rb:432:in `block in remote_specs'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/index.rb:9:in `build'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/source/rubygems.rb:431:in `remote_specs'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/source/rubygems.rb:131:in `specs'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/resolver.rb:167:in `index_for'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/resolver.rb:175:in `results_for'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/resolver.rb:112:in `search_for'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/resolver.rb:255:in `block in verify_gemfile_dependencies_are_found!'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/resolver.rb:252:in `map!'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/resolver.rb:252:in `verify_gemfile_dependencies_are_found!'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/resolver.rb:48:in `start'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/resolver.rb:23:in `resolve'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/definition.rb:269:in `resolve'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/definition.rb:180:in `resolve_remotely!'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/installer.rb:280:in `resolve_if_needed'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/installer.rb:82:in `block in run'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/process_lock.rb:19:in `rescue in lock'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/process_lock.rb:6:in `lock'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/installer.rb:71:in `run'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/installer.rb:23:in `install'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/cli/update.rb:78:in `run'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/cli.rb:303:in `block in update'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/settings.rb:131:in `temporary'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/cli.rb:302:in `update'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/cli.rb:31:in `dispatch'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/cli.rb:25:in `start'
  /usr/local/share/gems/gems/bundler-2.3.9/exe/bundle:48:in `block in <top (required)>'
  /usr/local/share/gems/gems/bundler-2.3.9/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
  /usr/local/share/gems/gems/bundler-2.3.9/exe/bundle:36:in `<top (required)>'
  /home/redmine/bin/bundle:23:in `load'
  /home/redmine/bin/bundle:23:in `<main>'
```

## Environment

```
Bundler       2.3.9
  Platforms   ruby, x86_64-linux
Ruby          2.5.9p229 (2021-04-05 revision 67939) [x86_64-linux]
  Full Path   /usr/bin/ruby
  Config Dir  /etc
RubyGems      2.7.6.3
  Gem Home    /usr/share/gems
  Gem Path    /home/redmine/.gem/ruby:/usr/share/gems:/usr/local/share/gems
  User Home   /home/redmine
  User Path   /home/redmine/.gem/ruby
  Bin Dir     /usr/bin
OpenSSL
  Compiled    OpenSSL 1.1.1k  FIPS 25 Mar 2021
  Loaded      OpenSSL 1.1.1k  FIPS 25 Mar 2021
  Cert File   /etc/pki/tls/cert.pem
  Cert Dir    /etc/pki/tls/certs
Tools
  Git         2.31.1
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2022-03-09
Git SHA           7f90c0083a
Released Version  true
```

## Bundler settings

```
build.pg
  Set for the current user (/home/redmine/.bundle/config): "--with-pg-config=/usr/pgsql-10.17/bin/pg_config"
without
  Set for your local app (/data/redmine/redmine-4.2.3_new/.bundle/config): [:development, :test]
  Set for the current user (/home/redmine/.bundle/config): [:development, :test, :rmagick]
```

### Gemfile

```ruby
source 'https://rubygems.org'

ruby '>= 2.4.0', '< 2.8.0'
gem 'bundler', '>= 1.12.0'

gem 'rails', '5.2.6'
gem 'sprockets', '~> 3.7.2' if RUBY_VERSION < '2.5'
gem 'globalid', '~> 0.4.2' if Gem.ruby_version < Gem::Version.new('2.6.0')
gem 'rouge', '~> 3.26.0'
gem 'request_store', '~> 1.5.0'
gem "mini_mime", "~> 1.0.1"
gem "actionpack-xml_parser"
gem 'roadie-rails', (RUBY_VERSION < '2.5' ? '~> 1.3.0' : '~> 2.2.0')
gem 'marcel'
gem "mail", "~> 2.7.1"
gem 'csv', (RUBY_VERSION < '2.5' ? ['>= 3.1.1', '<= 3.1.5'] : '~> 3.1.1')
gem 'nokogiri', (RUBY_VERSION < '2.5' ? '~> 1.10.0' : '~> 1.11.1')
gem 'i18n', '~> 1.8.2'
gem "rbpdf", "~> 1.20.0"
gem 'addressable'
gem 'rubyzip', '~> 2.3.0'

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :x64_mingw, :mswin]

# TOTP-based 2-factor authentication
gem 'rotp'
gem 'rqrcode'

# Optional gem for LDAP authentication
group :ldap do
  gem 'net-ldap', '~> 0.17.0'
end

# Optional gem for OpenID authentication
group :openid do
  gem "ruby-openid", "~> 2.9.2", :require => "openid"
  gem "rack-openid"
end

# Optional gem for exporting the gantt to a PNG file
group :minimagick do
  gem 'mini_magick', '~> 4.11.0'
end

# Optional Markdown support, not for JRuby
group :markdown do
  gem 'redcarpet', '~> 3.5.1'
end

# Include database gems for the adapters found in the database
# configuration file
require 'erb'
require 'yaml'
database_file = File.join(File.dirname(__FILE__), "config/database.yml")
if File.exist?(database_file)
  database_config = YAML::load(ERB.new(IO.read(database_file)).result)
  adapters = database_config.values.map {|c| c['adapter']}.compact.uniq
  if adapters.any?
    adapters.each do |adapter|
      case adapter
      when 'mysql2'
gem "mysql2", "~> 0.5.0", :platforms => [:mri, :mingw, :x64_mingw]
      when /postgresql/
gem "pg", "~> 1.2.2", :platforms => [:mri, :mingw, :x64_mingw]
      when /sqlite3/
gem "sqlite3", "~> 1.4.0", :platforms => [:mri, :mingw, :x64_mingw]
      when /sqlserver/
gem "tiny_tds", "~> 2.1.2", :platforms => [:mri, :mingw, :x64_mingw]
gem "activerecord-sqlserver-adapter", "~> 5.2.1", :platforms => [:mri, :mingw, :x64_mingw]
      else
warn("Unknown database adapter `#{adapter}` found in config/database.yml, use Gemfile.local to load your own database gems")
      end
    end
  else
    warn("No adapter found in config/database.yml, please configure it first")
  end
else
  warn("Please configure your config/database.yml first")
end

group :development do
  gem "yard"
end

group :test do
  gem "rails-dom-testing"
  gem 'mocha', '>= 1.4.0'
  gem 'simplecov', '~> 0.18.5', :require => false
  gem "ffi", platforms: [:mingw, :x64_mingw, :mswin]
  # For running system tests
  gem 'puma'
  gem 'capybara', '~> 3.31.0'
  gem "selenium-webdriver"
  gem 'webdrivers', '~> 4.4', require: false
  # RuboCop
  gem 'rubocop', '~> 1.12.0'
  gem 'rubocop-performance', '~> 1.10.1'
  gem 'rubocop-rails', '~> 2.9.0'
end

local_gemfile = File.join(File.dirname(__FILE__), "Gemfile.local")
if File.exists?(local_gemfile)
  eval_gemfile local_gemfile
end

# Load plugins' Gemfiles
Dir.glob File.expand_path("../plugins/*/{Gemfile,PluginGemfile}", __FILE__) do |file|
  eval_gemfile file
end
```

### plugins/redmine_contacts/Gemfile

```ruby
gem 'redmine_crm'
gem 'vcard', (RUBY_VERSION > '2.0' ? '>= 0' : '~> 0.2.16')
gem 'spreadsheet'
```

### plugins/redmine_checklists/Gemfile

```ruby
gem 'redmine_crm'
```

### plugins/redmine_agile/Gemfile

```ruby
gem "redmine_crm"
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (5.2.6)
      actionpack (= 5.2.6)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailer (5.2.6)
      actionpack (= 5.2.6)
      actionview (= 5.2.6)
      activejob (= 5.2.6)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (5.2.6)
      actionview (= 5.2.6)
      activesupport (= 5.2.6)
      rack (~> 2.0, >= 2.0.8)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionpack-xml_parser (2.0.1)
      actionpack (>= 5.0)
      railties (>= 5.0)
    actionview (5.2.6)
      activesupport (= 5.2.6)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.3)
    activejob (5.2.6)
      activesupport (= 5.2.6)
      globalid (>= 0.3.6)
    activemodel (5.2.6)
      activesupport (= 5.2.6)
    activerecord (5.2.6)
      activemodel (= 5.2.6)
      activesupport (= 5.2.6)
      arel (>= 9.0)
    activestorage (5.2.6)
      actionpack (= 5.2.6)
      activerecord (= 5.2.6)
      marcel (~> 1.0.0)
    activesupport (5.2.6)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 0.7, < 2)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    addressable (2.8.0)
      public_suffix (>= 2.0.2, < 5.0)
    arel (9.0.0)
    ast (2.4.2)
    builder (3.2.4)
    capybara (3.31.0)
      addressable
      mini_mime (>= 0.1.3)
      nokogiri (~> 1.8)
      rack (>= 1.6.0)
      rack-test (>= 0.6.3)
      regexp_parser (~> 1.5)
      xpath (~> 3.2)
    childprocess (3.0.0)
    chunky_png (1.4.0)
    concurrent-ruby (1.1.9)
    crass (1.0.6)
    css_parser (1.11.0)
      addressable
    csv (3.1.9)
    docile (1.4.0)
    erubi (1.10.0)
    globalid (0.4.2)
      activesupport (>= 4.2.0)
    htmlentities (4.3.4)
    i18n (1.8.11)
      concurrent-ruby (~> 1.0)
    liquid (2.6.3)
    loofah (2.13.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.1)
      mini_mime (>= 0.1.1)
    marcel (1.0.2)
    method_source (1.0.0)
    mini_magick (4.11.0)
    mini_mime (1.0.3)
    minitest (5.15.0)
    mocha (1.13.0)
    net-ldap (0.17.0)
    nio4r (2.5.8)
    nokogiri (1.11.7-x86_64-linux)
      racc (~> 1.4)
    parallel (1.21.0)
    parser (3.1.0.0)
      ast (~> 2.4.1)
    pg (1.2.3)
    public_suffix (4.0.6)
    puma (5.5.2)
      nio4r (~> 2.0)
    racc (1.6.0)
    rack (2.2.3)
    rack-openid (1.4.2)
      rack (>= 1.1.0)
      ruby-openid (>= 2.1.8)
    rack-test (1.1.0)
      rack (>= 1.0, < 3)
    rails (5.2.6)
      actioncable (= 5.2.6)
      actionmailer (= 5.2.6)
      actionpack (= 5.2.6)
      actionview (= 5.2.6)
      activejob (= 5.2.6)
      activemodel (= 5.2.6)
      activerecord (= 5.2.6)
      activestorage (= 5.2.6)
      activesupport (= 5.2.6)
      bundler (>= 1.3.0)
      railties (= 5.2.6)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.4.2)
      loofah (~> 2.3)
    railties (5.2.6)
      actionpack (= 5.2.6)
      activesupport (= 5.2.6)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.19.0, < 2.0)
    rainbow (3.0.0)
    rake (13.0.6)
    rbpdf (1.20.1)
      htmlentities
      rbpdf-font (~> 1.19.0)
    rbpdf-font (1.19.1)
    redcarpet (3.5.1)
    redmine_crm (0.0.55)
      liquid (< 2.6.4)
      rails
      rubyzip
    regexp_parser (1.8.2)
    request_store (1.5.0)
      rack (>= 1.4)
    rexml (3.2.5)
    roadie (4.0.0)
      css_parser (~> 1.4)
      nokogiri (~> 1.8)
    roadie-rails (2.2.0)
      railties (>= 5.1, < 6.2)
      roadie (>= 3.1, < 5.0)
    rotp (6.2.0)
    rouge (3.26.1)
    rqrcode (2.1.0)
      chunky_png (~> 1.0)
      rqrcode_core (~> 1.0)
    rqrcode_core (1.2.0)
    rubocop (1.12.1)
      parallel (~> 1.10)
      parser (>= 3.0.0.0)
      rainbow (>= 2.2.2, < 4.0)
      regexp_parser (>= 1.8, < 3.0)
      rexml
      rubocop-ast (>= 1.2.0, < 2.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (>= 1.4.0, < 3.0)
    rubocop-ast (1.15.1)
      parser (>= 3.0.1.1)
    rubocop-performance (1.10.2)
      rubocop (>= 0.90.0, < 2.0)
      rubocop-ast (>= 0.4.0)
    rubocop-rails (2.9.1)
      activesupport (>= 4.2.0)
      rack (>= 1.1)
      rubocop (>= 0.90.0, < 2.0)
    ruby-ole (1.2.12.2)
    ruby-openid (2.9.2)
    ruby-progressbar (1.11.0)
    rubyzip (2.3.2)
    selenium-webdriver (3.142.7)
      childprocess (>= 0.5, < 4.0)
      rubyzip (>= 1.2.2)
    simplecov (0.18.5)
      docile (~> 1.1)
      simplecov-html (~> 0.11)
    simplecov-html (0.12.3)
    spreadsheet (1.3.0)
      ruby-ole
    sprockets (4.0.2)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.4.2)
      actionpack (>= 5.2)
      activesupport (>= 5.2)
      sprockets (>= 3.0.0)
    thor (1.2.1)
    thread_safe (0.3.6)
    tzinfo (1.2.9)
      thread_safe (~> 0.1)
    unicode-display_width (2.1.0)
    vcard (0.3.0)
    webdrivers (4.6.1)
      nokogiri (~> 1.6)
      rubyzip (>= 1.3.0)
      selenium-webdriver (>= 3.0, < 4.0)
    webrick (1.7.0)
    websocket-driver (0.7.5)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.5)
    xpath (3.2.0)
      nokogiri (~> 1.8)
    yard (0.9.27)
      webrick (~> 1.7.0)

PLATFORMS
  x86_64-linux

DEPENDENCIES
  actionpack-xml_parser
  addressable
  bundler (>= 1.12.0)
  capybara (~> 3.31.0)
  csv (~> 3.1.1)
  ffi
  globalid (~> 0.4.2)
  i18n (~> 1.8.2)
  mail (~> 2.7.1)
  marcel
  mini_magick (~> 4.11.0)
  mini_mime (~> 1.0.1)
  mocha (>= 1.4.0)
  net-ldap (~> 0.17.0)
  nokogiri (~> 1.11.1)
  pg (~> 1.2.2)
  puma
  rack-openid
  rails (= 5.2.6)
  rails-dom-testing
  rbpdf (~> 1.20.0)
  redcarpet (~> 3.5.1)
  redmine_crm
  request_store (~> 1.5.0)
  roadie-rails (~> 2.2.0)
  rotp
  rouge (~> 3.26.0)
  rqrcode
  rubocop (~> 1.12.0)
  rubocop-performance (~> 1.10.1)
  rubocop-rails (~> 2.9.0)
  ruby-openid (~> 2.9.2)
  rubyzip (~> 2.3.0)
  selenium-webdriver
  simplecov (~> 0.18.5)
  spreadsheet
  tzinfo-data
  vcard
  webdrivers (~> 4.4)
  yard

RUBY VERSION
   ruby 2.5.9p229

BUNDLED WITH
   2.3.4
```

--- TEMPLATE END ----------------------------------------------------------------

Unfortunately, an unexpected error occurred, and Bundler cannot continue.

First, try this link to see if there are any existing issue reports for this error:
https://github.com/rubygems/rubygems/search?q=%28eval%29+2++syntax+error%2C+unexpected+%3D%3E%2C+expecting+%27%7D%27&type=Issues

If there aren't any reports for this error yet, please fill in the new issue form located at https://github.com/rubygems/rubygems/issues/new?labels=Bundler&template=bundler-related-issue.md, and copy and paste the report template above in there.

I even downloaded the redmine_crm-0.0.58.gem file and tried to install it manually, But also here i get an error:
```
$ gem install redmine_crm-0.0.58.gem
ERROR:  While executing gem ... (ArgumentError)
    wrong number of arguments (given 4, expected 1)
```

**Comment by deivid-rodriguez** [comment]
Hei @Lokizzle. None of the software you're using is supported anymore. I would start by trying to upgrade Ruby first.

Also, some of those messages seem to come from redmine itself, so you could try ask for help to Redline people.

Sorry I can't be of more help, this is probably something fixed in the latest Ruby/RubyGems/Bundler versions, so your best bet is to figure out how to upgrade.

Good luck!
